core/services/subscription.go

```diff
@@ -164,6 +164,12 @@ func runJob(store *strpkg.Store, le models.LogRequest, data models.JSON) {
 		input.SetError(err)
 		logger.Errorw(err.Error(), le.ForLogger()...)
 	}
+
+	if le.GetLog().Removed {
+		logger.Debugw("Skipping run for removed log", "log", le.GetLog(), "jobId", le.GetJobSpec().ID)
```

Comment:
Two ideas:

1. Can we place this higher up in the function? We're still doing a lot of work to throw away a log.
2. You could perform this check even higher up the stack, inside the `InitiatorSubscription#dispatchLog`: https://github.com/smartcontractkit/chainlink/blob/features/statspush-tx-hash-before-confirmed/core/services/subscription.go#L108-L118

I'm leaning towards 2.